chore: add maintainers handbook (#62)
aws-powertools/powertools-mcp: MAINTAINERS.md

# Maintainers Playbook

Playbook for active maintainers of the proejct and new contributors.

## Overview

> [!Note]
> This is a living document and will be updated as the project evolves.

> This is document explains who the maintainers are, their responsibilities, and how they should be doing it. If you're interested in contributing, see the [contributing](https://github.com/aws-powertools/powertools-mcp/blob/chore/main/CONTRIBUTING.md) document.

Reviewer: This document explains…

!!! note
    This won't change the CHANGELOG as the merge commit is immutable. Don't worry about it. We'd only rewrite git history only if this can lead to confusion and we'd pair with another maintainer.

**All looking good, what's next?**

The best part comes now. Replace the placeholder `[Human readable summary of changes]` with what you'd like to communicate to customers what this release is all about. Rule of thumb: always put yourself in the customers shoes.

These are some questions to keep in mind when drafting your first or future release notes:

- Can customers understand at a high level what changed in this release?
- Is there a link to the documentation where they can read more about each main change?
- Are there any graphics or [code snippets](https://carbon.now.sh/) that can enhance readability?
- Are we calling out any key contributor(s) to this release?

Once you're happy, hit `Publish release` 🎉🎉🎉.

This will kick off the post-release automations and within a few minutes you should see all issues labeled as `pending-release`  notified of the new release and labeled as `completed`.

### Maintain Overall Health of the Repo

> Keep the `main` branch at production quality at all times. Backport features as needed. Cut release branches and tags to enable future patches.

Reviewer: “Cut release branches and tags…” to “Create release branches and tags…”

### Manage Roadmap

See [Roadmap section](https://docs.powertools.aws.dev/lambda/typescript/latest/roadmap/)

Ensure the repo highlights features that should be elevated to the project roadmap. Be clear about the feature’s status, priority, target version, and whether or not it should be elevated to the roadmap.

### Add Continuous Integration Checks

Add integration checks that validate pull requests and pushes to ease the burden on Pull Request reviewers. Continuously revisit areas of improvement to reduce operational burden in all parties involved.

### Negative Impact on the Project

Actions that negatively impact the project will be handled by the admins, in coordination with other maintainers, in balance with the urgency of the issue. Examples would be [Code of Conduct](https://github.com/aws-powertools/powertools-mcp/blob/main/CODE_OF_CONDUCT.md)
{target="_blank"} violations, deliberate harmful or malicious actions, spam, monopolization, and security risks.

## Common scenarios

> These are recurring ambiguous situations that new and existing maintainers may encounter. They serve as guidance. It is up to each maintainer to follow, adjust, or handle in a different manner as long as [our conduct is consistent](#uphold-code-of-conduct)

Reviewer 2: ```suggestion
These are recurring ambiguous situations that new and existing maintainers may encounter. They serve as guidance. It is up to each maintainer to follow, adjust, or handle in an appropriate manner as long as [our conduct is consistent](#uphold-code-of-conduct)
```

### Contribution is stuck

A contribution can get stuck often due to lack of bandwidth and language barrier. For bandwidth issues, check whether the author needs help. Make sure you get their permission before pushing code into their existing PR - do not create a new PR unless strictly necessary.

> For language barrier and others, offer a 1:1 chat to get them unblocked. Often times, English might not be their primary language, and writing in public might put them off, or come across not the way they intended to be.

Reviewer 2: ```suggestion
In the case of a language barrier or other issues, offer a 1:1 chat to get them unblocked. Often times, English might not be their primary language, and writing in public might put them off, or come across not the way they intended to be.
```
